src/table_ops.jl

```diff
@@ -0,0 +1,307 @@
+
+# Determine the ordinality of a set of regularly spaced numerical coordinates
+function _coords_to_ords(
+    coords::AbstractVector{<:Real}, 
+    dim::Dimension, 
+    ::DimensionalData.Near,
+    position::DimensionalData.Position, 
+    span::DimensionalData.Regular)
+    step = DD.step(span)
+    float_ords = ((coords .- first(dim)) ./ step) .+ 1
+    int_ords = _round_ords(float_ords, position)
+    return clamp!(int_ords, 1, length(dim))
+end
+
+# Determine the ordinality of a set of categorical or irregular coordinates
+function _coords_to_ords(
+    coords::AbstractVector, 
+    dim::Dimension, 
+    sel::DimensionalData.Selector, 
+    ::DimensionalData.Position, 
+    ::DimensionalData.Span)
+    return map(c -> DimensionalData.selectindices(dim, rebuild(sel, c)), coords)
+end
+
+# Round coordinate ordinality to the appropriate integer given the specified locus
+_round_ords(ords::AbstractVector{<:Real}, ::DimensionalData.Start) = floor.(Int, ords)
+_round_ords(ords::AbstractVector{<:Real}, ::DimensionalData.Center) = round.(Int, ords)
+_round_ords(ords::AbstractVector{<:Real}, ::DimensionalData.End) = ceil.(Int, ords)
+
+# Extract dimension value from the given vector of coordinates
+_dim_vals(coords::AbstractVector, precision::Int) = _unique_vals(coords, precision)
+_dim_vals(coords::AbstractVector, ::DD.Order, precision::Int) = _unique_vals(coords, precision)
+_dim_vals(coords::AbstractVector, ::DD.ForwardOrdered, precision::Int) = sort!(_unique_vals(coords, precision))
+_dim_vals(coords::AbstractVector, ::DD.ReverseOrdered, precision::Int) = sort!(_unique_vals(coords, precision), rev=true)
+
+# Extract all unique coordinates from the given vector
+_unique_vals(coords::AbstractVector, precision::Int) = _round_dim_val.(coords, precision) |> unique
+
+# Round dimension value within the specified precision
+_round_dim_val(x, ::Int) = x
+_round_dim_val(x::Real, precision::Int) = round(x, digits=precision)
+
+# Determine if the given coordinates are forward ordered, reverse ordered, or unordered
+function _guess_dim_order(coords::AbstractVector)
+    if issorted(coords)
+        return DD.ForwardOrdered()
+    elseif issorted(coords, rev=true)
+        return DD.ReverseOrdered()
+    else
+        return DD.Unordered()
+    end
+end
+
+# Estimate the span between consecutive coordinates
+_guess_dim_span(::AbstractVector, ::DD.Order, ::Int) = DD.Irregular()
+_guess_dim_span(::AbstractVector{<:Real}, ::DD.Order, ::Int) = DD.Irregular()
+function _guess_dim_span(coords::AbstractVector{<:Real}, ::DD.Ordered, precision::Int)
+    steps = round.((@view coords[2:end]) .- (@view coords[1:end-1]), digits=precision)
+    span = argmin(abs, steps)
+    return all(isinteger, round.(steps ./ span, digits=precision)) ? DD.Regular(span) : DD.Irregular()
+end
+
+function _build_dim(vals::AbstractVector, dim::Symbol, order::DD.Order, ::DD.Span)
+    return Dim{dim}(DD.Categorical(vals, order=order))
```

Comment:
All of these should use `name2dim(dim)` instead of `Dim{dim}` to catch any `@dim` dimensions.

```suggestion
    return rebuild(name2dim(dim), DD.Categorical(vals, order=order))
```